rs/ledger_suite/icp/ledger/src/main.rs

```diff
@@ -1128,11 +1128,16 @@ fn account_balance_() {
 }
 
 #[candid_method(query, rename = "account_balance")]
-fn account_balance_candid_(arg: BinaryAccountBalanceArgs) -> Tokens {
-    let account = AccountIdentifier::from_address(arg.account).unwrap_or_else(|e| {
-        trap_with(&format!("Invalid account identifier: {}", e));
-    });
-    account_balance(account)
+fn account_balance_candid_(arg: AccountIdentifierByteBuf) -> Tokens {
+    match BinaryAccountBalanceArgs::try_from(arg) {
+        Ok(arg) => {
+            let account = AccountIdentifier::from_address(arg.account).unwrap_or_else(|e| {
+                trap_with(&format!("Invalid account identifier: {}", e));
+            });
+            account_balance(account)
+        }
+        Err(_) => Tokens::ZERO,
```

Comment:
Since this endpoint doesn't return a `Result`, and we already return `Tokens::ZERO` for account balance queries for accounts that do not exist, returning `Tokens::ZERO` also for the case where the `AccountIdentifier` blob is misformed seems a better option than returning a candid error.